frontend/server/src/Controllers/Problem.php

```diff
@@ -284,26 +284,17 @@ private static function validateCreateOrUpdate(
         $r->ensureInt('output_limit', 0, null, $isRequired);
         $r->ensureInt('input_limit', 0, null, $isRequired);
 
-        // HACK! I don't know why "languages" doesn't make it into $r, and I've spent far too much time
-        // on it already, so I'll just leave this here for now...
-        if (!isset($r['languages']) && isset($_REQUEST['languages'])) {
-            $languages = implode(',', $_REQUEST['languages']);
-        } elseif (isset($r['languages']) && is_array($r['languages'])) {
-            $languages = implode(',', $r['languages']);
-        } else {
-            $languages = strval($r['languages']);
-        }
+        $languages = $r->getStringList('languages', [], $isRequired);
         \OmegaUp\Validators::validateValidSubset(
-            $r['languages'],
+            $languages,
             'languages',
-            array_keys(\OmegaUp\Controllers\Run::SUPPORTED_LANGUAGES),
-            $isRequired
+            array_keys(\OmegaUp\Controllers\Run::SUPPORTED_LANGUAGES)
         );
 
         return [
             'problem' => $problem,
             'selectedTags' => $selectedTags,
-            'languages' => $languages,
+            'languages' => strval($r['languages']),
```

Comment:
aquí no sería

```suggestion
            'languages' => join(',', $r['languages']),
```

?

Reply:
No, ya de por sí `$r['languages']` contiene la forma `lenguaje1, lenguaje2, lenguaje3, ...`.

En todo caso, lo que puedo hacer es pasar `join(',', $languages)`

Reply:
ah sí! mejor haz eso